pkgs/development/python-modules/pysmi/default.nix

```
@@ -1,29 +1,61 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitHub,
+
+  # build-system
+  poetry-core,
+
+  # dependencies
   ply,
+  jinja2,
+  requests,
+
+  # tests
+  pysnmp,
+  pytestCheckHook,
 }:
 
-buildPythonPackage rec {
-  version = "0.3.4";
-  format = "setuptools";
-  pname = "pysmi";
+let
+  self = buildPythonPackage rec {
+    version = "1.4.4";
+    pname = "pysmi";
+    pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "bd15a15020aee8376cab5be264c26330824a8b8164ed0195bd402dd59e4e8f7c";
-  };
+    src = fetchFromGitHub {
+      owner = "lextudio";
+      repo = "pysmi";
+      rev = "refs/tags/v${version}";
+      hash = "sha256-9ArKo1UT4g+H8Z51NZ6rHlOhyz2grAc1V8Xl+ztfYic=";
+    };
+
+    build-system = [ poetry-core ];
+
+    dependencies = [
+      ply
+      jinja2
+      requests
+    ];
+
+    # Tests require pysnmp, which in turn requires pysmi => infinite recursion
+    doCheck = false;
+
+    nativeCheckInputs = [
+      pysnmp
+      pytestCheckHook
+    ];
 
-  propagatedBuildInputs = [ ply ];
+    pythonImportsCheck = [ "pysmi" ];
 
-  # Tests require pysnmp, which in turn requires pysmi => infinite recursion
-  doCheck = false;
+    passthru.tests.pytest = self.overridePythonAttrs { doCheck = true; };
```

```suggestion
    passthru.tests.pytest = pysmi.overridePythonAttrs { doCheck = true; };
```
wouldn't require as much refactoring